docs/contributing.rst

```diff
@@ -54,6 +54,30 @@ Python API documentation is automatically generated from the code using Sphinx's
 HTTP REST API documentation is automatically generated from the code using the
 `httpdomain <https://pythonhosted.org/sphinxcontrib-httpdomain/>`_ extension.
 
+Release notes
+-------------
+
+To add entries to the release notes, create a file in the ``news`` directory
+with the ``source.type`` name format, where ``type`` is one of:
+
+* ``feature``: for new features
+* ``bug``: for bug fixes
+* ``api``: for API changes
+* ``dev``: for development-related changes
+* ``author``: for contributor names
+* ``other``: for other changes
+
+And where the ``source`` part of the filename is:
+
+* ``42`` when the change is described in issue ``42``
+* ``PR42`` when the change has been implemented in pull request ``42``, and
+  there is no associated issue
+* ``Cabcdef`` when the change has been implemented in changeset ``abcdef``, and
+  there is no associated issue or pull request.
```

Comment:
Since you use mergify and it rebases, I recommend against using commit hashes - the author won't be able to predict their hash, and so any hashes they use will lead to a 404 link.

Reply:
Yeah, you are right. I didn't noticed this.